keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/DeliveryConfigRepositoryTests.kt

```diff
@@ -275,6 +275,29 @@ abstract class DeliveryConfigRepositoryTests<T : DeliveryConfigRepository, R : R
         )
       }
 
+
+      test("resources in an environment can be rechecked") {
+        // note: this test needs to be here even though it's testing a resource repository function
+        // because we need a valid config and environment for the resources to exist in, and those can only
+        // be saved with both a delivery config repository and a resource repository.
+        storeResources()
+        store()
+
+        val firstCheck = resourceRepository.itemsDueForCheck(Duration.ofMinutes(2), 4)
+        val secondCheck = resourceRepository.itemsDueForCheck(Duration.ofMinutes(2), 4)
+        resourceRepository.triggerResourceRecheck("test", deliveryConfig.application)
+        val afterRecheck = resourceRepository.itemsDueForCheck(Duration.ofMinutes(2), 4)
+        val testResources = deliveryConfig.environments.find { it.name == "test" }?.resourceIds ?: emptySet()
+
+        expect {
+          that(firstCheck.size).isEqualTo(4)
+          that(secondCheck.size).isEqualTo(0)
+          that(afterRecheck.size).isEqualTo(2)
```

Comment:
I'm guessing the count here is 2 because there's more than one environment in the delivery config and you're only resetting "test", is that it? Might be worth adding a comment.

Reply:
Yep, there are two envs with two resources each. Do comments on tests help you a lot? (trying to gauge when it is worth it to add comments vs when it just makes the tests cluttered. I'll add here but I'm curious your general thoughts about when you yourself would add comments to tests)

Reply:
In cases like these, when it's not obvious where certain values are coming from, or around logic that might not be self-explanatory, yes, I tend to add a comment. Actually I don't think of test code that much differently from "main" code. It's code that we have to maintain, and we actually have more test code than application code, so in these specific cases I find it worth the trouble/noise. 🤷‍♂️ 

Example from my open PR:
https://github.com/spinnaker/keel/blob/99cc244e387cc64165aac061714ba4d728fa4ab3/keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/DeliveryConfigRepositoryTests.kt#L494-L495

https://github.com/spinnaker/keel/blob/99cc244e387cc64165aac061714ba4d728fa4ab3/keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/DeliveryConfigRepositoryTests.kt#L746